src/grpc/zero_copy_stream.cc

```diff
@@ -65,9 +68,40 @@ bool GrpcZeroCopyInputStream::Next(const void** data, int* size) {
 void GrpcZeroCopyInputStream::BackUp(int count) {
   if (0 < count && static_cast<size_t>(count) <= position_) {
     position_ -= count;
+    bytes_read_ -= count;
   }
 }
 
+bool GrpcZeroCopyInputStream::Skip(int count) {
+  if (count < 0) {
+    // Safe guard against wrong usage.
+    return false;
+  }
+  size_t count_left = static_cast<size_t>(count);
+  while (position_ + count_left > current_buffer_size_) {
+    // Skipping past the current buffer, read the next one.
+    int delta = static_cast<int>(current_buffer_size_ - position_);
+    count_left -= delta;
+    bytes_read_ += delta;
+    position_ = 0;
+    if (!serializer_.Next(&current_buffer_, &current_buffer_size_)) {
+      // No data. We are potentially not at the end of the stream yet, but we
+      // don't know that and can only skip to the end and return an error.
+      current_buffer_size_ = 0;
+      return false;
+    }
+  }
+
+  // Move the position ahead the requested number of bytes.
+  position_ += count_left;
+  bytes_read_ += count_left;
+  return true;
+}
+
+::google::protobuf::int64 GrpcZeroCopyInputStream::ByteCount() const {
```

Comment:
Thanks for implementing this too. Just curious, do you know where this used/needed? I'm assuming it's not related to originally reported error.

Reply:
Oh, I see it's part of the contract for `Skip`.

```
  // Skips a number of bytes.  Returns false if the end of the stream is
  // reached or some input error occurred.  In the end-of-stream case, the
  // stream is advanced to the end of the stream (so ByteCount() will return
  // the total size of the stream).
  virtual bool Skip(int count) = 0;

  // Returns the total number of bytes read since this object was created.
  virtual int64_t ByteCount() const = 0;
```

Reply:
Ack